### PR TITLE
Change `Po` to `C` with some exceptions

### PR DESCRIPTION
**Pull request by kojiishi**
This patch implements the proposal at:
https://github.com/kojiishi/unicode-auto-spacing/issues/11#issuecomment-1968365524

### CODE REVIEW

File: auto-spacing.txt [hunk 1]
```diff
@@ -686,9 +701,14 @@ FD50..FD8F     ; N  # N   ARABIC LIGATURE TEH WITH JEEM WITH MEEM INITIAL FORM..
 FD92..FDC7     ; N  # N   ARABIC LIGATURE MEEM WITH JEEM WITH KHAH INITIAL FORM..ARABIC LIGATURE NOON WITH JEEM WITH YEH FINAL FORM
 FDF0..FDFB     ; N  # N   ARABIC LIGATURE SALLA USED AS KORANIC STOP SIGN ISOLATED FORM..ARABIC LIGATURE JALLAJALALOUHOU
 FE00..FE0F     ; N  # A   VARIATION SELECTOR-1..VARIATION SELECTOR-16
-FE10..FE19     ; O  # W   PRESENTATION FORM FOR VERTICAL COMMA..PRESENTATION FORM FOR VERTICAL HORIZONTAL ELLIPSIS
+FE10..FE16     ; N  # W   PRESENTATION FORM FOR VERTICAL COMMA..PRESENTATION FORM FOR VERTICAL QUESTION MARK
```
Comment (kojiishi) [inline auto-spacing.txt:704]: @xfq This one looks like an oversight, no? Should we exclude EAW=W?

File: auto-spacing.txt [hunk 2]
```diff
@@ -866,29 +884,31 @@ FFFD           ; O  # A   REPLACEMENT CHARACTER
...
 16B7D..16B8F   ; N  # N   PAHAWH HMONG CLAN SIGN TSHEEJ..PAHAWH HMONG CLAN SIGN VWJ
 16E40..16E7F   ; N  # N   MEDEFAIDRIN CAPITAL LETTER M..MEDEFAIDRIN SMALL LETTER Y
+16E97..16E9A   ; N  # N   MEDEFAIDRIN COMMA..MEDEFAIDRIN EXCLAMATION OH
 16F00..16F4A   ; N  # N   MIAO LETTER PA..MIAO LETTER RTE
 16F4F..16F87   ; N  # N   MIAO SIGN CONSONANT MODIFIER BAR..MIAO VOWEL SIGN UI
 16F8F..16F9F   ; N  # N   MIAO TONE RIGHT..MIAO LETTER REFORMED TONE-8
 16FE0..16FE1   ; W  # W   TANGUT ITERATION MARK..NUSHU ITERATION MARK
-16FE2          ; O  # W   OLD CHINESE HOOK MARK
-16FE3          ; N  # W   OLD CHINESE ITERATION MARK
+16FE2..16FE3   ; N  # W   OLD CHINESE HOOK MARK..OLD CHINESE ITERATION MARK
```
Comment (kojiishi) [inline auto-spacing.txt:911]: @xfq This one too, EAW=W.

File: auto-spacing.txt [hunk 3]
```diff
@@ -7,19 +7,27 @@
 #
 # @missing: 0000..10FFFF; O
 
-0030..0039     ; N  # Na  DIGIT ZERO..DIGIT NINE
-0041..005A     ; N  # Na  LATIN CAPITAL LETTER A..LATIN CAPITAL LETTER Z
+0021..0023     ; N  # Na  EXCLAMATION MARK..NUMBER SIGN
```
Comment (kojiishi) [inline auto-spacing.txt:10]: Shouldn't [U+0022 QUOTATION MARK](https://codepoints.net/U+0022) be `O` when curly quotes are `O`?

Example: The "永" Chinese character represents "forever".

File: auto-spacing.txt [hunk 4]
```diff
@@ -7,19 +7,27 @@
...
+0025..0027     ; N  # Na  PERCENT SIGN..APOSTROPHE
```
Comment (kojiishi) [inline auto-spacing.txt:11]: Should [U+0027 APOSTROPHE](https://codepoints.net/U+0027) be the same category as QUOTATION MARK?

File: auto-spacing.txt (hunk 3 above)
Comment (kojiishi) [reply auto-spacing.txt:10]: Maybe this is no issue if we can limit this only when the content language is Chinese; i.e., not applicable to English.

File: auto-spacing.txt (hunk 4 above)
Comment (xfq) [reply auto-spacing.txt:11]: Agreed.

File: auto-spacing.txt [hunk 5]
```diff
@@ -414,16 +413,29 @@
...
 2018..2019     ; O  # A   LEFT SINGLE QUOTATION MARK..RIGHT SINGLE QUOTATION MARK
 201C..201D     ; O  # A   LEFT DOUBLE QUOTATION MARK..RIGHT DOUBLE QUOTATION MARK
-2020..2022     ; O  # A   DAGGER..BULLET
-2024..2027     ; O  # A   ONE DOT LEADER..HYPHENATION POINT
-2030           ; O  # A   PER MILLE SIGN
-2032..2033     ; O  # A   PRIME..DOUBLE PRIME
-2035           ; O  # A   REVERSED PRIME
-203B           ; O  # A   REFERENCE MARK
-203E           ; O  # A   OVERLINE
+2020..2022     ; N  # A   DAGGER..BULLET
```
[the remark this answers is not in the document]
Comment (xfq) [reply auto-spacing.txt:421]: Agreed.

File: auto-spacing.txt (hunk 1 above)
Comment (xfq) [reply auto-spacing.txt:704]: I think we should, yes.

File: auto-spacing.txt [hunk 6]
```diff
@@ -7,19 +7,30 @@
 #
 # @missing: 0000..10FFFF; O
 
+0021           ; CN  # Na  EXCLAMATION MARK
```
Comment (xfq) [inline auto-spacing.txt:10]: I think we should document what `CN` represents in the proposal.

File: auto-spacing.txt [hunk 7]
```diff
@@ -413,16 +478,30 @@
...
-2035           ; O  # A   REVERSED PRIME
-203B           ; O  # A   REFERENCE MARK
-203E           ; O  # A   OVERLINE
+2020..2021     ; O  # A   DAGGER..DOUBLE DAGGER
+2022           ; C  # A   BULLET
+2023           ; C  # N   TRIANGULAR BULLET
+2024..2025     ; C  # A   ONE DOT LEADER..TWO DOT LEADER
+2026           ; O  # A   HORIZONTAL ELLIPSIS
+2027           ; C  # A   HYPHENATION POINT
+2030           ; C  # A   PER MILLE SIGN
```
Comment (kojiishi) [inline auto-spacing.txt:492]: @xfq @kidayasuo U+2030 is in cp932, cp936, cp949. Is this ok to be `C`?

File: auto-spacing.txt [hunk 8]
```diff
@@ -413,16 +478,30 @@
...
+2031           ; C  # N   PER TEN THOUSAND SIGN
+2032..2033     ; C  # A   PRIME..DOUBLE PRIME
+2034           ; C  # N   TRIPLE PRIME
+2035           ; C  # A   REVERSED PRIME
+2036..2038     ; C  # N   REVERSED DOUBLE PRIME..CARET
+203B           ; C  # A   REFERENCE MARK
+203C..203D     ; C  # N   DOUBLE EXCLAMATION MARK..INTERROBANG
```
Comment (kojiishi) [inline auto-spacing.txt:499]: @kidayasuo @xfq These are in cp932 (Shift-JIS), is `C` ok?
U+203C DOUBLE EXCLAMATION MARK

File: auto-spacing.txt [hunk 9]
```diff
@@ -413,16 +478,30 @@
...
+203E           ; C  # A   OVERLINE
+2041..2043     ; C  # N   CARET INSERTION POINT..HYPHEN BULLET
+2047..2051     ; C  # N   DOUBLE QUESTION MARK..TWO ASTERISKS ALIGNED VERTICALLY
```
Comment (kojiishi) [inline auto-spacing.txt:502]: @kidayasuo @xfq These are in cp932 (Shift-JIS), is `C` ok?
U+2047 DOUBLE QUESTION MARK
U+2048 QUESTION EXCLAMATION MARK
U+2049 EXCLAMATION QUESTION MARK

File: auto-spacing.txt [hunk 10]
```diff
@@ -413,16 +478,30 @@
...
-203E           ; O  # A   OVERLINE
+2020..2021     ; O  # A   DAGGER..DOUBLE DAGGER
+2022           ; C  # A   BULLET
+2023           ; C  # N   TRIANGULAR BULLET
+2024..2025     ; C  # A   ONE DOT LEADER..TWO DOT LEADER
+2026           ; O  # A   HORIZONTAL ELLIPSIS
+2027           ; C  # A   HYPHENATION POINT
+2030           ; C  # A   PER MILLE SIGN
+2031           ; C  # N   PER TEN THOUSAND SIGN
+2032..2033     ; C  # A   PRIME..DOUBLE PRIME
```
Comment (kojiishi) [inline auto-spacing.txt:494]: @xfq @kidayasuo These are in cp932, 936, and 949. Is `C` ok?

File: auto-spacing.txt [hunk 11]
```diff
@@ -413,16 +478,30 @@
...
-2024..2027     ; O  # A   ONE DOT LEADER..HYPHENATION POINT
-2030           ; O  # A   PER MILLE SIGN
-2032..2033     ; O  # A   PRIME..DOUBLE PRIME
-2035           ; O  # A   REVERSED PRIME
-203B           ; O  # A   REFERENCE MARK
-203E           ; O  # A   OVERLINE
+2020..2021     ; O  # A   DAGGER..DOUBLE DAGGER
+2022           ; C  # A   BULLET
+2023           ; C  # N   TRIANGULAR BULLET
+2024..2025     ; C  # A   ONE DOT LEADER..TWO DOT LEADER
```
Comment (kojiishi) [inline auto-spacing.txt:489]: @xfq @kidayasuo U+2025 is in cp932, 949, 950. Ok to be `C`?

**Comment by kojiishi** [comment]
@xfq @kidayasuo FYI, I created a list of legacy encoding, for the range of U+2010-204F.
Code    Char    EA      GC      EAW     cp932   sjis04  cp936   cp949   cp950   Name
DEBUG:UnicodeDataEntry:Missing entry: @missing: 0000..10FFFF; N
U2010   ‐       A       Pd      A       815D    815D    A95C                    HYPHEN
U2011   ‑       N       Pd      N                                               NON-BREAKING HYPHEN
U2012   ‒       N       Pd      N                                               FIGURE DASH
U2013   –       A       Pd      A               829C    A843            A156    EN DASH
U2014   —       A       Pd      A                       A1AA            A158    EM DASH
U2015   ―       A       Pd      A       815C    815C    A844    A1AA            HORIZONTAL BAR
U2016   ‖       A       Po      A       8161    8161    A1AC                    DOUBLE VERTICAL LINE
U2017   ‗       N       Po      N                                               DOUBLE LOW LINE
U2018   ‘       A       Pi      A       8165    8165    A1AE    A1AE    A1A5    LEFT SINGLE QUOTATION MARK
U2019   ’       A       Pf      A       8166    8166    A1AF    A1AF    A1A6    RIGHT SINGLE QUOTATION MARK
U201A   ‚       N       Ps      N                                               SINGLE LOW-9 QUOTATION MARK
U201B   ‛       N       Pi      N                                               SINGLE HIGH-REVERSED-9 QUOTATION MARK
U201C   “       A       Pi      A       8167    8167    A1B0    A1B0    A1A7    LEFT DOUBLE QUOTATION MARK
U201D   ”       A       Pf      A       8168    8168    A1B1    A1B1    A1A8    RIGHT DOUBLE QUOTATION MARK
U201E   „       N       Ps      N                                               DOUBLE LOW-9 QUOTATION MARK
U201F   ‟       N       Pi      N                                               DOUBLE HIGH-REVERSED-9 QUOTATION MARK
U2020   †       A       Po      A       81F5    81F5            A2D3            DAGGER
U2021   ‡       A       Po      A       81F6    81F6            A2D4            DOUBLE DAGGER
U2022   •       A       Po      A               825F                    A145    BULLET
U2023   ‣       N       Po      N                                               TRIANGULAR BULLET
U2024   ․       A       Po      A                                               ONE DOT LEADER
U2025   ‥       A       Po      A       8164    8164    A845    A1A5    A14C    TWO DOT LEADER
U2026   …       A       Po      A       8163    8163    A1AD    A1A6    A14B    HORIZONTAL ELLIPSIS
U2027   ‧       A       Po      A                                       A145    HYPHENATION POINT
U2028           N       Zl      N                                               LINE SEPARATOR
U2029           N       Zp      N                                               PARAGRAPH SEPARATOR
U202A   ‪       N       Cf      N                                               LEFT-TO-RIGHT EMBEDDING
U202B   ‫       N       Cf      N                                               RIGHT-TO-LEFT EMBEDDING
U202C   ‬       N       Cf      N                                               POP DIRECTIONAL FORMATTING
U202D   ‭       N       Cf      N                                               LEFT-TO-RIGHT OVERRIDE
U202E   ‮       N       Cf      N                                               RIGHT-TO-LEFT OVERRIDE
U202F           N       Zs      N                                               NARROW NO-BREAK SPACE
U2030   ‰       A       Po      A       81F1    81F1    A1EB    A2B6            PER MILLE SIGN
U2031   ‱       N       Po      N                                               PER TEN THOUSAND SIGN
U2032   ′       A       Po      A       818C    818C    A1E4    A1C7    A1AC    PRIME
U2033   ″       A       Po      A       818D    818D    A1E5    A1C8            DOUBLE PRIME
U2034   ‴       N       Po      N                                               TRIPLE PRIME
U2035   ‵       A       Po      A                       A846            A1AB    REVERSED PRIME
U2036   ‶       N       Po      N                                               REVERSED DOUBLE PRIME
U2037   ‷       N       Po      N                                               REVERSED TRIPLE PRIME
U2038   ‸       N       Po      N                                               CARET
U2039   ‹       N       Pi      N                                               SINGLE LEFT-POINTING ANGLE QUOTATION MARK
U203A   ›       N       Pf      N                                               SINGLE RIGHT-POINTING ANGLE QUOTATION MARK
U203B   ※       A       Po      A       81A6    81A6    A1F9    A1D8    A1B0    REFERENCE MARK
U203C   ‼       N       Po      N               84E9                            DOUBLE EXCLAMATION MARK
U203D   ‽       N       Po      N                                               INTERROBANG
U203E   ‾       A       Po      A               007E                    A1C2    OVERLINE
U203F   ‿       N       Pc      N               8677                            UNDERTIE
U2040   ⁀       N       Pc      N                                               CHARACTER TIE
U2041   ⁁       N       Po      N                                               CARET INSERTION POINT
U2042   ⁂       N       Po      N               86FC                            ASTERISM
U2043   ⁃       N       Po      N                                               HYPHEN BULLET
U2044   ⁄       N       Sm      N                                               FRACTION SLASH
U2045   ⁅       N       Ps      N                                               LEFT SQUARE BRACKET WITH QUILL
U2046   ⁆       N       Pe      N                                               RIGHT SQUARE BRACKET WITH QUILL
U2047   ⁇       N       Po      N               84EA                            DOUBLE QUESTION MARK
U2048   ⁈       N       Po      N               84EB                            QUESTION EXCLAMATION MARK
U2049   ⁉       N       Po      N               84EC                            EXCLAMATION QUESTION MARK
U204A   ⁊       N       Po      N                                               TIRONIAN SIGN ET
U204B   ⁋       N       Po      N                                               REVERSED PILCROW SIGN
U204C   ⁌       N       Po      N                                               BLACK LEFTWARDS BULLET
U204D   ⁍       N       Po      N                                               BLACK RIGHTWARDS BULLET
U204E   ⁎       N       Po      N                                               LOW ASTERISK
U204F   ⁏       N       Po      N                                               REVERSED SEMICOLON

**Comment by kojiishi** [comment]
Updated the sheet to include `C` and legacy encodings, FYI.

**Pull request merged**